core/metacling/src/TCling.cxx

```diff
@@ -5891,11 +5867,106 @@ Int_t TCling::AutoParse(const char *cls)
    return nHheadersParsed > 0 ? 1 : 0;
 }
 
+static void* LazyFunctionCreatorAutoloadForModule(const std::string& mangled_name,
+      cling::Interpreter *fInterpreter) {
+   using namespace llvm::object;
+   using namespace llvm::sys::path;
+   using namespace llvm::sys::fs;
+
+   static constexpr std::hash<std::string> hash_fn;
+   // size_t is a hashed mangled_name and libs[(uint8_t)mname_lib_table.second] = library's
+   // name in string.
+   static std::unordered_map<size_t, uint8_t> mname_lib_table;
+   static std::vector<std::string> libs;
+   // Initialize library table at first run
+   static bool fFirstRun = true;
+
+   R__LOCKGUARD(gInterpreterMutex);
+
+   // Initialize the table
+   if (fFirstRun) {
+      clang::Preprocessor &PP = fInterpreter->getCI()->getPreprocessor();
+      const HeaderSearchOptions &HSOpts = PP.getHeaderSearchInfo().getHeaderSearchOpts();
+      auto ModulePaths(HSOpts.PrebuiltModulePaths);
+
+      // sort and eliminate dupicate path
+      std::sort(ModulePaths.begin(), ModulePaths.end());
```

Comment:
Can we run a PCM-based ROOT with a rootmap based experiment framework, or do we nee to wait with adoption until everything is module-based?